### PR TITLE
[Container app] Fix #28983: `az containerapp job registry`: Support commands for modifying the job's registries

### LINKED ISSUE #28983: Support updating registry settings in az containerapp job update
**Related command**
az containerapp job update

**Is your feature request related to a problem? Please describe.**
There seems to be a cyclic dependency when creating a containerapp job using an image from ACR with managed identity authentication. The managed identity of the job needs to exist before an AcrPull permission can be given to it. If you try to configure the ACR in the registry settings (ie. using Terraform) the creation will fail (assuming that the job will verify the pull on the image, for which it does not have permissions yet). Therefore the registry settings need to be set after the AcrPull permission has been given to the managed identity. I therefore need to set the registry settings, pointing to the ACR, after initial creation. "az container app update" allows thisi, but this option does not seem to exist for "az containerapp job update". 

**Describe the solution you'd like**
Registry settings parameters on "az containerapp job update".

**Describe alternatives you've considered**
No good alternatived exist.

**Additional context**

### PR DESCRIPTION
**Pull request by Greedygre**
**Related command**

Fix issue: https://github.com/Azure/azure-cli/issues/28983
Support command `az containerapp job registry set/remove/show/list`.

**Description**

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Container app

File: src/azure-cli/azure/cli/command_modules/containerapp/tests/latest/test_containerappjob_registry.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+# pylint: skip-file
+
+
+class ContainerAppJobsRegistryOperationsTest(ScenarioTest):
+    # def __init__(self, *arg, **kwargs):
+    #     super().__init__(*arg, random_config_dir=True, **kwargs)
+
+    @ResourceGroupPreparer(location="northeurope")
+    @AllowLargeResponse()
+    def test_containerapp_registry_curd(self, resource_group):
```
Comment (anandanthony) [inline src/azure-cli/azure/cli/command_modules/containerapp/tests/latest/test_containerappjob_registry.py:29]: should this be `test_containerapp_registry_crud`?

Comment (Greedygre) [reply src/azure-cli/azure/cli/command_modules/containerapp/tests/latest/test_containerappjob_registry.py]: Fixed.

**Review by anandanthony — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/containerapp/commands.py [hunk 2]
```diff
@@ -72,6 +72,12 @@ def load_command_table(self, _):
         g.custom_command('remove', 'remove_managed_identity_job', confirmation=True, supports_no_wait=True, exception_handler=ex_handler_factory())
         g.custom_show_command('show', 'show_managed_identity_job')
 
+    with self.command_group('containerapp job registry') as g:
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/containerapp/commands.py:75]: I would recommend to add preview flag when first add the command group

**Review by Juliehzl — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/containerapp/_params.py [hunk 3]
```diff
@@ -476,3 +476,12 @@ def load_arguments(self, _):
 
     with self.argument_context('containerapp job identity remove') as c:
         c.argument('user_assigned', nargs='*', help="Space-separated user identities. If no user identities are specified, all user identities will be removed.")
+
+    with self.argument_context('containerapp job registry') as c:
+        c.argument('server', help="The container registry server, e.g. myregistry.azurecr.io")
+        c.argument('username', help='The username of the registry. If using Azure Container Registry, we will try to infer the credentials if not supplied')
+        c.argument('password', help='The password of the registry. If using Azure Container Registry, we will try to infer the credentials if not supplied')
+        c.argument('identity', help="The managed identity with which to authenticate to the Azure Container Registry (instead of username/password). Use 'system' for a system-defined identity or a resource id for a user-defined identity. The managed identity should have been assigned acrpull permissions on the ACR before deployment (use 'az role assignment create --role acrpull ...').")
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/containerapp/_params.py:484]: Please refer to this doc [managed_identity_command_guideline](https://github.com/Azure/azure-cli/blob/dev/doc/managed_identity_command_guideline.md) to design the managed identity related parameters.

File: src/azure-cli/azure/cli/command_modules/containerapp/commands.py [hunk 4]
```diff
@@ -72,6 +72,12 @@ def load_command_table(self, _):
         g.custom_command('remove', 'remove_managed_identity_job', confirmation=True, supports_no_wait=True, exception_handler=ex_handler_factory())
         g.custom_show_command('show', 'show_managed_identity_job')
 
+    with self.command_group('containerapp job registry', is_preview=True) as g:
+        g.custom_command('set', 'set_registry_job', exception_handler=ex_handler_factory())
+        g.custom_show_command('show', 'show_registry_job')
+        g.custom_command('list', 'list_registry_job')
+        g.custom_command('remove', 'remove_registry_job', exception_handler=ex_handler_factory())
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/containerapp/commands.py:79]: Do you need to consider adding `confirmation=True` for the remove command?

Comment (Greedygre) [reply src/azure-cli/azure/cli/command_modules/containerapp/commands.py:79]: This command is for remove registry under containerapp, registry is part of configuration in Containerapp's property.
Consistent with other this kind of command, not need `confirmation=True`.

File: src/azure-cli/azure/cli/command_modules/containerapp/_params.py (hunk 3 above)
Comment (Greedygre) [reply src/azure-cli/azure/cli/command_modules/containerapp/_params.py:484]: This `--identity` is keep same as `az containerapp registry set`, the `--identity` allows `system` and resource id for managed identity.

File: src/azure-cli/azure/cli/command_modules/containerapp/commands.py (hunk 4 above)
Comment (zhoxing-ms) [reply src/azure-cli/azure/cli/command_modules/containerapp/commands.py:79]: OK, got it

File: src/azure-cli/azure/cli/command_modules/containerapp/_params.py (hunk 3 above)
Comment (Greedygre) [reply src/azure-cli/azure/cli/command_modules/containerapp/_params.py:484]: Hi @zhoxing-ms 
 About this, we consider keep consistent with existing commands `az containerapp registry set --indentity`. We will consider to make them align with the design and deprecate in future, thanks.

Comment (zhoxing-ms) [reply src/azure-cli/azure/cli/command_modules/containerapp/_params.py:484]: OK, sounds good!

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**